### DISCUSSION TITLE
Is it possible to use a custom listing within an html table via an ejs template with sort and filter option?

### DISCUSSION THREAD

**Discussion by MindMosaicI** [quarto-dev/quarto-cli#9402] [Q&A]
### Description

Dear all, 

I want to use a custom listing with sort and filter options within an html table (created with an ejs template). However, I have not yet succeeded in integrating the quarto internal sort and filter options. But for our project it is absolutely necessary to integrate these functions. I have tried various customizations in the ejs and html file, but it does not seem to be possible. If we use the normal listing via a table, it works. Is there a way to use these functions in a custom listing with an html table via an ejs template? 

Thanks for your help

**Comment by mcanouil** [comment]
Could you explain a bit more of your use case?

Any reason to not use the default table listings? See <https://quarto.org/docs/websites/website-listings.html#listing-types>.

**Reply by MindMosaicI** [reply]
Of course, 
a usecase would be that we list several products in the table, with different data in different columns (one product per row), but the filters and sort option allow only one filter to be applied at a time. For us, however, it would be necessary to be able to select several filters or sorting options. For example, sort by size and price. So that the cheapest and largest product is listed in row 1. 
Or, in the case of a list of people, that you can filter by both first name and surname at the same time. The number of filter and sorting options to be selected would be defined individually in each case. 
However, this cannot be set using the Quarto functions, so we would want to control the whole thing using the ejs template. However, it is not clear how this can be integrated into the Quarto website or implemented within Quarto. 
I have tried using DataTables, but this is not displayed via Quarto.

Thanks for your help

**Reply by mcanouil** [reply]
I don't think a custom listings is the solution here.
DataTables is available as wrappers in many languages such as Python and R.
I've seen examples of those.
To me, what you want is what DataTable does.

> I have tried using DataTables, but this is not displayed via Quarto.

I don't know what that means without actual code.
